sgm/inference/helpers.py

```diff
@@ -0,0 +1,519 @@
+import os
```

Comment:
if you add these helpers to sgm, we should remove them from https://github.com/Stability-AI/generative-models/blob/main/scripts/demo/streamlit_helpers.py

and adapt https://github.com/Stability-AI/generative-models/blob/main/scripts/demo/sampling.py

otherwise this looks duplicated

Reply:
this file can be cleaned but not the goal of the PR

Reply:
It will maybe take quite some additional effort to clean up the duplication cf. streamlit helpers, since those versions use streamlit's widget functions in conditional blocks, e.g.

https://github.com/Stability-AI/generative-models/blob/e25e4c0df1d01fb9720f62c73b4feab2e4003e3f/scripts/demo/streamlit_helpers.py#L249-L252

only uses/shows that widget when that selection is done, and expressing that in a way that lets one use these inference helpers with streamlit doesn't quite sound straightforward.

If you're going to defer cleaning this file up, it's probably not worth to look at that duplication right now either :)

Reply:
To @akx's point it is more complicated, but that may mean the helpers need to change in some meaningful way and it might be better to deal with it now to avoid breaking changes once these are live, since the idea is for people to use them. I will take a crack at it this morning, if it gets too hairy we can still wait.

Reply:
For a repo whose version is 0.0.1 I don't think some possible future breaking changes in helpers are that bad 😁

Reply:
> if you add these helpers to sgm, we should remove them from https://github.com/Stability-AI/generative-models/blob/main/scripts/demo/streamlit_helpers.py
> 
> and adapt https://github.com/Stability-AI/generative-models/blob/main/scripts/demo/sampling.py
> 
> otherwise this looks duplicated

This is for the most part done. There are new enums and classes in `api.py` that are not used by streamlit (yet), which avoids rewriting the UI code. But almost all inference code is now shared between `api.py` and `sampling.py` through the `helpers.py` module.